Dragonflight/WarriorArms.lua

```diff
@@ -499,12 +500,22 @@ local TriggerCollateralDamage = setfenv( function()
     collateralDmgStacks = 0
 end, state )
 
+local TriggerCrushingAdvance = setfenv( function()
+    addStack( "crushing_advance" )
+end, state )
+
+
 spec:RegisterCombatLogEvent( function( _, subtype, _,  sourceGUID, sourceName, _, _, destGUID, destName, destFlags, _, spellID, spellName, _, _, _, _, critical_swing, _, _, critical_spell )
     if sourceGUID == state.GUID then
         if subtype == "SPELL_CAST_SUCCESS" then
             if ( spellName == class.abilities.colossus_smash.name or spellName == class.abilities.warbreaker.name ) then
                 last_cs_target = destGUID
             end
+        -- Tier30 4pc Set Bonus check
+        elseif subtype == "SPELL_DAMAGE" and state.set_bonus.tier30_4pc > 0 then
+            local ability = class.abilities[ spellID ]
+            if not ability then return end
+            if ability.key == "deep_wounds" and critical_spell then TriggerCrushingAdvance() end
```

Comment:
We don't need to manually apply auras for actual reality.  We only need to apply auras in `reset_precast` if those auras won't be detected when auras are scraped (i.e., a fake or hidden aura or some other kind of tracking).

Reply:
So in this case TriggerCrushingAdvance() isn't needed on line 503 and the CLEU portion isn't needed either, and the handler within MS will still work? That would simplifies crushing advance dramatically.

Reply:
Correct.  If that's what procs a visible buff on the character, you don't have to tell the addon to apply it.  The addon will find the aura since it actually exists.

If you need to predict something, like a mechanic that occurs on every other attack, on the next swing, has an internal cooldown, etc., then there can be a need to track the data elements to make that prediction.  I don't think that's the case here.